cmd/add.go

```diff
@@ -1,54 +1,155 @@
 // Copyright (c) 2023 Cisco Systems, Inc. and its affiliates
 // All rights reserved.
-
 package cmd
 
 import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
 	"github.com/cisco-open/grabit/internal"
+	"github.com/rs/zerolog/log"
 	"github.com/spf13/cobra"
 )
 
 func addAdd(cmd *cobra.Command) {
 	addCmd := &cobra.Command{
-		Use:   "add",
+		Use:   "add [url]",
 		Short: "Add new resource",
-		Args:  cobra.MinimumNArgs(1),
+		Args:  cobra.ExactArgs(1),
 		RunE:  runAdd,
 	}
+
 	addCmd.Flags().String("algo", internal.RecommendedAlgo, "Integrity algorithm")
 	addCmd.Flags().String("filename", "", "Target file name to use when downloading the resource")
 	addCmd.Flags().StringArray("tag", []string{}, "Resource tags")
+	addCmd.Flags().String("cache", "", "URL of Artifactory cache")
+
 	cmd.AddCommand(addCmd)
 }
 
 func runAdd(cmd *cobra.Command, args []string) error {
+	// Get flags
 	lockFile, err := cmd.Flags().GetString("lock-file")
 	if err != nil {
-		return err
-	}
-	lock, err := internal.NewLock(lockFile, true)
-	if err != nil {
-		return err
+		return fmt.Errorf("failed to get lock-file flag: %w", err)
```

Comment:
Why are you adding all these `fmt.Errorf()` statements?  These do not have anything to do with the functionality of this PR.

If you'd like to propose changing this functionality, great -- do so!  But do it in a different PR.

Please remove all these types of changes from this PR.  This PR should *only* contain the functionality for uploading to / downloading from Artifactory.